### PR TITLE
Commission HUD

### PR DESCRIPTION
**Pull request by TacoMonkey11**
Only commissions, has only been tested by 1 other person

### CODE REVIEW

**Review by ExternalTime — APPROVED** [review]
There are few cosmetic issues, but it definitely works.

File: src/main/java/me/xmrvizzy/skyblocker/config/SkyblockerConfig.java [hunk 1]
```diff
@@ -123,6 +123,17 @@ public static class DwarvenMines {
...
+        public DwarvenHud dwarvenHud = new DwarvenHud();
+    }
+
+    public static class DwarvenHud {
+        public boolean enabled = true;
+        public boolean enableBackground = true;
+        @ConfigEntry.BoundedDiscrete(min = 3, max = 700)
+        public int x = 10;
+        @ConfigEntry.BoundedDiscrete(min = 3, max = 427)
+        public int y = 10;
```
[the remark this answers is not in the document]
Comment (TacoMonkey11) [reply src/main/java/me/xmrvizzy/skyblocker/config/SkyblockerConfig.java:136]: I guess I could just raise the limit to 2k each

File: src/main/java/me/xmrvizzy/skyblocker/skyblock/dwarven/DwarvenHud.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+                            if (matcher.find()) {
+                                commissions.add(new Commission(matcher.group(1), matcher.group(2)));
+                            }
+
+                        }
+                    }
+                });
+                if (commissions.size() > 0){
+                    if (SkyblockerConfig.get().locations.dwarvenMines.dwarvenHud.enableBackground)
+                        DrawableHelper.fill(matrixStack, hudX, hudY, hudX + 150, hudY + (20 * commissions.size()), 0x64000000);
```
[the remark this answers is not in the document]
Comment (TacoMonkey11) [reply src/main/java/me/xmrvizzy/skyblocker/skyblock/dwarven/DwarvenHud.java:53]: Raised it to 200 pixels

File: src/main/java/me/xmrvizzy/skyblocker/skyblock/dwarven/DwarvenHud.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+            Pattern.compile("^.*((?:Lava Springs|Cliffside Veins|Rampart's Quarry|Upper Mines|Royal Mines) Mithril): (\\d+\\.?\\d*%|DONE)"),
+            Pattern.compile("^.*((?:Lava Springs|Cliffside Veins|Rampart's Quarry|Upper Mines|Royal Mines) Titanium): (\\d+\\.?\\d*%|DONE)"),
+            Pattern.compile("^.*(Goblin Raid): (\\d+\\.?\\d*%|DONE)"),
+            Pattern.compile("^.*((?:Powder Ghast|Star Sentry) Puncher): (\\d+\\.?\\d*%|DONE)"),
+            Pattern.compile("^.*((?<!Lucky )Raffle): (\\d+\\.?\\d*%|DONE)"),
+            Pattern.compile("^.*(Lucky Raffle): (\\d+\\.?\\d*%|DONE)"),
+            Pattern.compile("^.*(2x Mithril Powder Collector): (\\d+\\.?\\d*%|DONE)"),
+            Pattern.compile("^.*((?:Ruby|Amber|Sapphire|Jade|Amethyst|Topaz) Gemstone Collector): (\\d+\\.?\\d*%|DONE)"),
+            Pattern.compile("^.*((?:Amber|Sapphire|Jade|Amethyst|Topaz) Crystal Hunter): (\\d+\\.?\\d*%|DONE)"),
+            Pattern.compile("^.*(Chest Looter): (\\d+\\.?\\d*%|DONE)")
```
[the remark this answers is not in the document]
Comment (ExternalTime) [reply src/main/java/me/xmrvizzy/skyblocker/skyblock/dwarven/DwarvenHud.java]: Ok, I should have been more explicit.
I didn't mean that you should be compiling regexes in a loop. What I meant is that you can construct List of regexes from a list of strings. You can either assign COMMISSIONS from a static block or you could do something like this:
```
COMMISSIONS = List.of(
    "(?:Titanium|Mithril|Hard Stone) Miner",
    ...)
    .stream()
    .map(x -> Pattern.compile("(" + x + "): (\\d+\\.?\\d*%|DONE)"))
    .collect(Collectors.toList());
```

[the remark this answers is not in the document]
Comment (TacoMonkey11) [reply src/main/java/me/xmrvizzy/skyblocker/skyblock/dwarven/DwarvenHud.java]: ok I implemented that, and now it they don't match, even when they clearly do:

regex: `((?:Titanium|Mithril|Hard Stone) Miner): (\d+\.?\d*%|DONE)`

string: ` Titanium Miner: 20%`

**Pull request merged**